app/models/mixins/authentication_mixin.rb

```diff
@@ -158,6 +158,8 @@ def update_authentication(data, options = {})
       cred.userid = value[:userid]
       cred.password = value[:password]
       cred.auth_key = value[:auth_key]
+      # TODO(lwander) this should have its own cred attribute
+      cred.password = value[:json] unless value[:json].blank?
```

Comment:
Sure.  Looks like @matthewd is pointing you [in the right direction](https://github.com/ManageIQ/manageiq/pull/5208#discussion_r44083374).